feat: Add Nats

johanfol opened the pull request:
Adding Nats as a startup option;
Create Factory to trigger different startup servers; Add test coverage reporting for baseline.

johanfol committed:
feat: Add Nats

Adding Nats as a startup option;
Create Factory to trigger different startup servers;
Add test coverage reporting for baseline.

JeanPierreNell commented on src/index.ts:
```diff
@@ -1,4 +1,32 @@
+        this.startupService = new JetstreamService();
+        break;
+      case 'nats':
+        this.startupService = new JetstreamService();
```
This should spin up the NATS Service.

johanfol commented:
https://github.com/frmscoe/General-Issues/issues/133

johanfol committed:
bug: Fix promise issue

Fix promise issue where Jetstream would not wait for Producer to startup before carrying on;
Refactor index.ts to remove the factory from there and rather be its own class;
Rename SERVER_TYPE to STARTUP_TYPE as server type seems to be reserved for NodeJS.

JeanPierreNell reviewed (approved):
Outstanding Issue withstanding , LGTM

johanfol committed:
fix: closed the stream early for Nats specifically

johanfol committed:
lint: Linting

johanfol committed:
feat: Bump Version of Lib
